### DISCUSSION TITLE
How do i obtain a last inserted ID while inserting?

### DISCUSSION THREAD

**Discussion by afidegnum** [devashishdxt/rexie#20] [Q&A] [answered]
While creating an insert, I would like to obtain the last inserted id record which I can pass to the dom. 

What's the function needed?

**Comment by devashishdxt** [comment, marked as answer]
Seems like you're using `auto_increment` IDs. If you go in the readme, you can see that [`Store::add`](https://docs.rs/rexie/latest/rexie/struct.Store.html#method.add) returns the ID of the added object.

Here is the example from readme:

```rust
use rexie::*;

async fn add_employee(rexie: &Rexie, name: &str, email: &str) -> Result<u32> {
    // Create a new read-write transaction
    let transaction = rexie.transaction(&["employees"], TransactionMode::ReadWrite)?;

    // Get the `employees` store
    let employees = transaction.store("employees")?;

    // Create an employee
    let employee = serde_json::json!({
        "name": name,
        "email": email,
    });
    // Convert it to `JsValue`
    let employee = serde_wasm_bindgen::to_value(&employee).unwrap();

    // Add the employee to the store
    let employee_id = employees.add(&employee, None).await?;

    // Waits for the transaction to complete
    transaction.done().await?;

    // Return the employee id
    Ok(num_traits::cast(employee_id.as_f64().unwrap()).unwrap())
}
```

**Reply by afidegnum** [reply]
Thanks a lot,

**Reply by afidegnum** [reply]
and how do i get the result associated with the ID? I initially listed the existing records. and there is a button click which supposed to add a record and return the inserted record in order to append the existing list.

**Reply by afidegnum** [reply]
ok, ive gotten it from one of your examples
